awx/sso/pipeline.py

```diff
@@ -77,6 +77,21 @@ def _update_m2m_from_expression(user, related, expr, remove=True):
         related.remove(user)
 
 
+def get_or_create_with_default_galaxy_cred(**kwargs):
+    from awx.main.models import Organization, Credential
+
+    (org, org_created) = Organization.objects.get_or_create(**kwargs)
+    if org_created:
+        logger.debug("Created org {} (id {}) from {}".format(org.name, org.id, kwargs))
+        public_galaxy_credential = Credential.objects.filter(managed=True, name='Ansible Galaxy').first()
+        if public_galaxy_credential is not None and public_galaxy_credential not in org.galaxy_credentials.all():
+            org.galaxy_credentials.add(public_galaxy_credential)
+            logger.debug("Added default Ansible Galaxy credential to org")
+        else:
```

Comment:
You might want to modify the `else:` to `elif public_galaxy_credential is not None:`

Because otherwise, the log isn't very clear. It _could_ find the default Ansible Galaxy credential, but it is already associated with the organization. If this is the case, then I would suggest that no log would be preferable, because no action has been taken, and no anomaly has happened.

Reply:
I also thought about just taking that check off because we only get in here if we created an org. So this would only be triggered if someone altered the creation of an org to auto-add the galaxy credential.

Reply:
Oh I see that now. So you might be able to delete `and public_galaxy_credential not in org.galaxy_credentials.all()`, because you have more information than the old method, which is that it's a new organization, so the galaxy_credential list should necessarily be empty. That would be better performance-wise as well.

Reply:
Removed.